### DISCUSSION TITLE
How to add a property with Templater?

### DISCUSSION THREAD

**Discussion by igorteuri** [SilentVoid13/Templater#1450] [Help] [answered]
Hi there!
My situation is: I have a system with folders and subfolders, for instance:
Mother folder
|__ Child folder 1
|__ Child folder 2

I wanted all the folder within Mother folder to have a tag or property associated with the Mother folder.

I have managed to rename the title using this code:

```javascript
<%*  
const hasTitle = !tp.file.title.startsWith("Untitled");  
let title;

let currentFolderPath = tp.file.folder(true);
let parentFolder = currentFolderPath.split("/").slice(-2, -1)[0];

if (!hasTitle) {  
    title = await tp.system.prompt("Número da Sessão");  
    
    await tp.file.rename("Sessão " + title + " - " + parentFolder);  
} else {  
    title = tp.file.title;  
}  
_%>
```

But I couldn't find a function to add a property or tag to the note. 

I was searching for something like:

```javascript 
   await tp.file.appendYalm("campanha",  parentFolder);  
```

Can someone help me? Is there a way to do that?

**Comment by Zachatoo** [comment, marked as answer]
There's not a Templater function for that, but you can use Obsidian APIs. Here's an example from the [documentation](https://silentvoid13.github.io/Templater/internal-functions/internal-modules/hooks-module.html).

```js
<%*
tp.hooks.on_all_templates_executed(async () => {
  const file = tp.file.find_tfile(tp.file.path(true));
  await app.fileManager.processFrontMatter(file, (frontmatter) => {
    frontmatter["key"] = "value";
  });
});
-%>
```

**Reply by igorteuri** [reply]
It worked like a charm! Although my current knowlegde at js doesn't me allow to fully understand how it worked hahaha 
Thanks a lot
